### PR TITLE
Docs: clarify correct example in no-return-await (fixes #13656)

### LINKED ISSUE #13656: Potentially incorrect documentation example for no-return-await
According to the docs: https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md the following are not essentially equivalent.

Example of invalid usage:
```
async function foo() {
    return await bar();
}
```
Example of valid usage:
```
async function foo() {
    const x = await bar();
    return x;
}
```
That makes no sense. Neither is proper usage. Is this a documentation bug or am I misunderstanding why it's there (stack trace is all I can think of)

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

fixes #13656

#### What changes did you make? (Give an overview)

* Added a comment above an example that could be misinterpreted.
* Also added missing eslint directives.

### CODE REVIEW

**Review by btmills — APPROVED** [review]

File: docs/rules/no-return-await.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ This rule aims to prevent a likely common performance hazard due to a lack of un
 Examples of **incorrect** code for this rule:
 
 ```js
+/*eslint no-return-await: "error"*/
```
Comment (anikethsaha) [inline docs/rules/no-return-await.md:14]: [no-await-in-loop](https://github.com/eslint/eslint/blob/master/docs/rules/no-await-in-loop.md) doesn't have this directive as well. 
It looks like we are having this in almost all rules, so maybe we can fix that in this PR or may be in a separate PR as well.

**Review by anikethsaha — CHANGES_REQUESTED** [review]
One optional thought

File: docs/rules/no-return-await.md (hunk 1 above)
Comment (mdjermanovic) [reply docs/rules/no-return-await.md:14]: Nice find! I made PR #13673 to fix that.

Comment (anikethsaha) [reply docs/rules/no-return-await.md:14]: Thanks

**Review by anikethsaha — APPROVED** [review]
